pipeline.Dockerfile

```diff
@@ -52,6 +52,4 @@ RUN update-alternatives --install /usr/bin/python python /usr/bin/python3 1 \
     && vagrant plugin install vagrant-libvirt \
     # Install Kubernetes collections
     && pip install --no-compile --no-cache-dir kubernetes \
-    && ansible-galaxy collection install kubernetes.core \
-    # Clean cache python
-    && find / -type d -name '*__pycache__' -prune -exec rm -rf {} \;
+    && ansible-galaxy collection install kubernetes.core
```

Comment:
If you plan to keep this in this PR as well maybe you could consider to bundle this PR with this one https://github.com/kubernetes-sigs/kubespray/pull/10017?

Reply:
I think this is fair. Why don't we use this one instead and dump the layer caching PR since you're ok with it. I was a bit afraid that the PR was going to get too large.